doc/using_xgboost.rst

```diff
@@ -0,0 +1,200 @@
+Hyperparameters are passed to your script as arguments and can be retrieved with an argparse.ArgumentParser instance.
+
+For a complete example of an XGBoost training script, see https://github.com/awslabs/amazon-sagemaker-examples/blob/master/introduction_to_amazon_algorithms/xgboost_abalone/abalone.py.
+
+Let's look at the main elements of the script. Starting with the ``__main__`` guard,
+use a parser to read the hyperparameters passed to the estimator when creating the training job.
+These hyperparameters are made available as arguments to our input script.
+We also parse a number of Amazon SageMaker-specific environment variables to get information about the training environment,
+such as the location of input data and location where we want to save the model.
+
+.. code:: python
+
+    if __name__ == '__main__':
+        parser = argparse.ArgumentParser()
+
+    # Hyperparameters are described here
+    parser.add_argument('--num_round', type=int)
+    parser.add_argument('--max_depth', type=int, default=5)
+    parser.add_argument('--eta', type=float, default=0.2)
+    parser.add_argument('--objective', type=str, default='reg:squarederror')
+    
+    # Sagemaker specific arguments. Defaults are set in the environment variables.
+    parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAIN'])
+    parser.add_argument('--validation', type=str, default=os.environ['SM_CHANNEL_VALIDATION'])
+    
+    args = parser.parse_args()
+    
+    train_hp = {
+        'max_depth': args.max_depth,
+        'eta': args.eta,
+        'gamma': args.gamma,
+        'min_child_weight': args.min_child_weight,
+        'subsample': args.subsample,
+        'silent': args.silent,
+        'objective': args.objective
+    }
+    
+    dtrain = xgb.DMatrix(args.train)
+    dval = xgb.DMatrix(args.validation)
+    watchlist = [(dtrain, 'train'), (dval, 'validation')] if dval is not None else [(dtrain, 'train')]
+
+    callbacks = []
+    prev_checkpoint, n_iterations_prev_run = add_checkpointing(callbacks)
+    # If checkpoint is found then we reduce num_boost_round by previously run number of iterations
+    
+    bst = xgb.train(
+        params=train_hp,
+        dtrain=dtrain,
+        evals=watchlist,
+        num_boost_round=(args.num_round - n_iterations_prev_run),
+        xgb_model=prev_checkpoint,
+        callbacks=callbacks
+    )
+    
+    model_location = args.model_dir + '/xgboost-model'
+    pkl.dump(bst, open(model_location, 'wb'))
+    logging.info("Stored trained model at {}".format(model_location))
+
+In the training script, you can customize the inference behavior by implementing the follwing functions:
+* ``input_fn`` - how input data is handled.
+* ``predict_fn`` - how the model is invokedfunction, and how the response is returned ).
+* ``output_fn`` - How the response data is handled
+
+These functions are optional. If you want to use the default implementations, do not implement them in your training script.
```

Comment:
I think it's a little strange to talk about inference functions in a _training_ script - maybe move this to the "Deploy" section

Reply:
I can move it to the Deploy section, but I've talked about a "training script". Should we call it something else, or suggest that you could put it in a separate "inference script"?

Reply:
I think I might go with something like "in your inference script, which can be in the same file as your training script from above"